src/autowiring/test/AutoConfigTest.cpp

```diff
@@ -157,3 +186,51 @@ TEST_F(AutoConfigTest, ContextSetAfter) {
   AutoRequired<MyConfigurableClass> mcc;
   ASSERT_EQ(mcc->b, 10442);
 }
+
+namespace {
+  class slider
+  {
+  public:
+    template<typename U>
+    struct valid {
+      static const bool value = std::is_arithmetic<U>::value;
+    };
+  };
+
+  static_assert(aw::valid<float, slider>::value, "Float metadata field not detected as being valid");
+  static_assert(!std::is_arithmetic<std::string>::value, "Strings are not arithmetic");
+  static_assert(!slider::valid<std::string>::value, "Slider should not believe strings are valid");
+  static_assert(!aw::valid<std::string, slider>::value, "String metadata field incorrectly detected as being valid");
+
+  class SliderManager {
+  public:
+    SliderManager(void) {
+      AutoCurrentContext ctxt;
+      ctxt->Config.WhenFn(
+        [this] (const aw::config_event&, const slider& slider) {
+          sliderReg.push_back(slider);
+        }
+      );
+    }
+
+    std::vector<slider> sliderReg;
+  };
+
+  class ClassWithBoundsField {
+  public:
+    std::atomic<int> b{ 929 };
+
+    static autowiring::config_descriptor GetConfigDescriptor(void) {
+      return{
+        aw::config_field { "b", "Field B description", &ClassWithBoundsField::b, 929, slider{}, aw::bounds<int>{ 10, 423 } },
+      };
+    }
+  };
+}
+
+TEST_F(AutoConfigTest, OpenRegistry) {
```

Comment:
Suggest renaming the test to "WhenFn" since that is the only functionality tested here.